src/librustdoc/html/highlight.rs

```diff
@@ -401,19 +473,24 @@ impl<'a> Classifier<'a> {
                 sink(Highlight::Token { text, class: None });
                 return;
             }
-            TokenKind::Ident => match get_real_ident_class(text, self.edition) {
-                Class::Ident => match text {
+            TokenKind::Ident => match get_real_ident_class(text, self.edition, false) {
+                None => match text {
                     "Option" | "Result" => Class::PreludeTy,
                     "Some" | "None" | "Ok" | "Err" => Class::PreludeVal,
                     _ if self.in_macro_nonterminal => {
                         self.in_macro_nonterminal = false;
                         Class::MacroNonTerminal
                     }
-                    _ => Class::Ident,
+                    "self" | "Self" => {
+                        Class::Self_(self.new_span(before, before + text.len() as u32))
+                    }
+                    _ => Class::Ident(self.new_span(before, before + text.len() as u32)),
```

Comment:
Preexisting but this looks a bit brittle :-/

Reply:
I just realized that it could be much better by doing:

```rust
self.new_span(before, text);
```

So then in the function we create the correct `hi`. Less operations, easier to read. :)